core/domain/interaction_jobs_one_off.py

```diff
@@ -392,3 +393,56 @@ def get_invalid_values(value_type, value, choices):
     @staticmethod
     def reduce(key, values):
         yield (key, values)
+
+
+class LogicProofInteractionOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that produces a list of (exploaration, state) pairs that use the
+    logicProof interaction. This will be used to send manual emails.
+    This job can be deleted after logicProof is removed from the codebase.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        owner_ids_empty = False
+        found_user_email = False
+        user_email = None
+        for state in exploration.states.values():
+            if found_user_email or owner_ids_empty:
+                break
```

Comment:
Why is this needed?

Reply:
Removed owner_ids_empty. This makes sure that states in no longer iterated upon, in case we find the email.